core/src/main/scala/com/metamx/tranquility/beam/ClusteredBeam.scala

```diff
@@ -132,7 +133,40 @@ class ClusteredBeam[EventType: Timestamper, InnerBeamType <: Beam[EventType]](
   private[this] val rand = new Random
 
   // Merged beams we are currently aware of, interval start millis -> merged beam.
-  private[this] val beams = ConcurrentMap[Long, Beam[EventType]]()
+  private[this] var beams: List[Beam[EventType]] =
+  {
+    try {
+      val dataPath = zpathWithDefault("data", ClusteredBeamMeta.empty.toBytes(objectMapper))
+      curator.sync().forPath(dataPath)
+      val zkMetaData = ClusteredBeamMeta.fromBytes(objectMapper, curator.getData.forPath(dataPath)).fold(
+        e => {
+          emitAlert(e, log, emitter, WARN, "Failed to read beam data from cache: %s" format identifier, alertMap)
+          throw e
+        },
+        meta => meta
+      )
+      log.info("Synced from ZK - [%s]", zkMetaData)
+      (zkMetaData.beamDictss map {
+        beamDicts =>
+          beamMergeFn(
+            beamDicts._2.zipWithIndex map {
+              case (beamDict, partitionNum) =>
+                val decorate = beamDecorateFn(
+                  new Interval(beamDict.get("interval").get),
+                  partitionNum
+                )
+                decorate(beamMaker.fromDict(beamDict))
+            }
+          )
+      }).toList sortBy (-_.getInterval().get.start.millis)
+    }
+    catch {
+      case e: Throwable =>
+        // Log Throwables to avoid invisible errors caused by https://github.com/twitter/util/issues/100.
+        log.error(e, "Failed to sync with zookeeper: %s", identifier)
+        throw e
```

Comment:
looks like https://github.com/twitter/util/issues/100 has been fixed, makes sense to let the process die